openmrs/src/test/java/org/motechproject/openmrs/tasks/OpenMRSTaskDataProviderTest.java

```diff
@@ -500,6 +506,33 @@ public void shouldReturnEmptyListWhenProgramEnrollmentNotFoundForLookupByMotechI
         assertEquals(0, actual.getNumberOfPrograms());
     }
 
+    @Test
+    public void shouldReturnBahmniProgramEnrollmentForPatientMotechIdAndProgramName() {
+        String className = BAHMNI_PROGRAM_ENROLLMENT;
+
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(PATIENT_MOTECH_ID, DEFAULT_MOTECH_ID);
+        lookupFields.put(PROGRAM_NAME, PROGRAM_DEFAULT_NAME);
+
+        List<ProgramEnrollment> expected = prepareProgramEnrollments(true);
+        when(programEnrollmentService.getBahmniProgramEnrollmentByPatientMotechId(eq(CONFIG_NAME), eq(DEFAULT_MOTECH_ID)))
+                .thenReturn(prepareProgramEnrollments(true));
+
+        Object object = taskDataProvider.lookup(className + '-' + CONFIG_NAME, BY_MOTECH_ID_AND_PROGRAM_NAME, lookupFields);
+
+        verify(programEnrollmentService).getBahmniProgramEnrollmentByPatientMotechId(eq(CONFIG_NAME), eq(DEFAULT_MOTECH_ID));
+
+        assertTrue(object instanceof ProgramEnrollmentListResult);
+
+        ProgramEnrollmentListResult actual = (ProgramEnrollmentListResult) object;
+
+        assertEquals(expected.get(0), actual.getResults().get(0));
```

Comment:
we need to check also if ProgramEnrollmentListResult contains only active/not active programs